### PR TITLE
bitnami/redis Add namespaceOverride capability to redis chart

### PR DESCRIPTION
**Pull request by jgmartinez**
### Description of the change

Similarly as in other charts, this change adds the namespaceOverride variable to Redis, allowing to deploy the chart in a different namespace as the helm installation default. This is really useful when the chart is installed as a dependency of another chart.

### Benefits

Redis can now be deployed in a different namespace as the indicated in the Helm installation command, if needed.

### Possible drawbacks

Namespace in namespaceOverride needs to be created prior installation, due to Helm's limitation on creating namespaces for subcharts.

### Applicable issues

No open issues

### Additional information

This replaces #21365

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by jgmartinez** [comment]
@fmulero Moved here

**Review by fmulero — APPROVED** [review]
Thanks a lot Juan for your contribution.

LGTM

**Review by fmulero — CHANGES_REQUESTED** [review]
Sorry again @jgmartinez

I've just figure out that some special characters have been included in your changes instead of white spaces. Please replace all of them with a single white space

File: bitnami/redis/templates/headless-svc.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ apiVersion: v1
 kind: Service
 metadata:
   name: {{ printf "%s-headless" (include "common.names.fullname" .) }}
-  namespace: {{ .Release.Namespace | quote }}
+  namespace: {{ include "common.names.namespace" . | quote }}
```
[the remark this answers is not in the document]
Comment (jgmartinez) [reply bitnami/redis/templates/headless-svc.yaml]: That was weird, should be fine now 😄

[the remark this answers is not in the document]
Comment (fmulero) [reply bitnami/redis/templates/headless-svc.yaml]: Great! thanks

**Review by fmulero — APPROVED** [review]
LGTM

**Pull request merged**